### PR TITLE
Relation mode changes for GA

### LINKED ISSUE #4029: Docs for relation mode GA
Document changes for GA:
- Remove preview mentions
- More coming later

### PR DESCRIPTION
**Pull request by keerlu**
## Describe this PR

- Remove mentions of `referentialIntegrity` preview feature
- Add more detail about indexes

## What issue does this fix?

Fixes #4029

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx [hunk 1]
```diff
@@ -98,21 +98,6 @@ For relational databases, the available options are:
...
-  provider     = "mysql"
-  url          = env("DATABASE_URL")
-  relationMode = "prisma"
-}
-
-generator js {
-  provider        = "prisma-client-js"
-  previewFeatures = ["referentialIntegrity"]
-}
-```
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx:114]: This is the one that should probably replaced with a mention between which versions the preview feature flag was necessary for later reference.

**Review by janpio — APPROVED** [review]

File: content/300-guides/050-database/850-using-prisma-with-planetscale.mdx [hunk 2]
```diff
@@ -69,19 +69,14 @@ If you try to push to a production branch, you will get the [error message](/ref
 
 PlanetScale does not allow foreign keys in its database schema. By default, Prisma uses foreign keys in the underlying database to enforce relations between fields in your Prisma schema. In Prisma versions 3.1.1 and later, you can [emulate relations in Prisma Client with the `prisma` relation mode](/concepts/components/prisma-schema/relations/relation-mode#emulate-relations-in-prisma-with-the-prisma-relation-mode), which avoids the need for foreign keys in the database.
 
-The ability to set the relation mode is currently a [preview feature](/concepts/components/preview-features). To enable this, add it to the `previewFeatures` list in the `generator` block of `schema.prisma`, and then set the `relationMode` field to `"prisma"` in the `datasource` block:
+To enable emulation of relations in Prisma Client, set the `relationMode` field to `"prisma"` in the `datasource` block:
```
Comment (janpio) [inline content/300-guides/050-database/850-using-prisma-with-planetscale.mdx:72]: I think the "needed a preview feature flag from vesion x to y" would fit into the admonition below quite nicely

File: content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx (hunk 1 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx:114]: I've updated the admonition to have the version info

File: content/300-guides/050-database/850-using-prisma-with-planetscale.mdx (hunk 2 above)
Comment (keerlu) [reply content/300-guides/050-database/850-using-prisma-with-planetscale.mdx:72]: Done

**Pull request merged**